### PR TITLE
[Drop-In UI] Add unit tests for missing classes (Other)

### PR DESCRIPTION
**Pull request by tomaszrybakiewicz**
Closes NAVAND-704

### Description
- Added unit tests for ReplayRouteTripSession
- Added unit-tests for NavigationView.
- Introduced factory methods for components and coordinators created by NavigationView.

### CODE REVIEW

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt [hunk 1]
```diff
@@ -124,30 +128,24 @@ class NavigationView @JvmOverloads constructor(
...
+                scalebarPlaceholderCoordinator(binding.scalebarLayout),
+                maneuverCoordinator(binding.guidanceLayout),
+                infoPanelCoordinator(binding.infoPanelLayout, binding.guidelineBottom),
+                actionButtonsCoordinator(binding.actionListLayout),
+                speedLimitCoordinator(binding.speedLimitLayout),
+                roadNameCoordinator(binding.roadNameLayout),
+                leftFrameCoordinator(binding.emptyLeftContainer),
+                rightFrameCoordinator(binding.emptyRightContainer)
+            )
+        }
```
Comment (tomaszrybakiewicz) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt:148]: Unfortunately, I was unable to test the `MapboxNavigationApp.setup()` call.
For some reason, Mockk doesn't detect the matcher for `MapboxNavigationApp.setup(NavigationOptions)`:

```
mockkObject(MapboxNavigationApp)
every { MapboxNavigationApp.setup(ofType(NavigationOptions::class)) } returns MapboxNavigationApp
```

When ran, it returns an error:
```
Failed matching mocking signature for
SignedCall(retValue=, isRetValueMock=true, retType=class com.mapbox.navigation.core.lifecycle.MapboxNavigationAppDelegate, self=MapboxNavigationApp(object MapboxNavigationApp), method=getMapboxNavigationAppDelegate(), args=[], invocationStr=MapboxNavigationApp(object MapboxNavigationApp).getMapboxNavigationAppDelegate())
SignedCall(retValue=, isRetValueMock=true, retType=class com.mapbox.navigation.core.lifecycle.MapboxNavigationAppDelegate, self=, method=setup(NavigationOptionsProvider), args=[com.mapbox.navigation.core.lifecycle.MapboxNavigationApp$$Lambda$220/0x0000000800c3d840@5e2c3382], invocationStr=MapboxNavigationAppDelegate(temporary mock).setup(com.mapbox.navigation.core.lifecycle.MapboxNavigationApp$$Lambda$220/0x0000000800c3d840@5e2c3382))
left matchers: [ofType(NavigationOptions)]
io.mockk.MockKException: Failed matching mocking signature for
SignedCall(retValue=, isRetValueMock=true, retType=class com.mapbox.navigation.core.lifecycle.MapboxNavigationAppDelegate, self=MapboxNavigationApp(object MapboxNavigationApp), method=getMapboxNavigationAppDelegate(), args=[], invocationStr=MapboxNavigationApp(object MapboxNavigationApp).getMapboxNavigationAppDelegate())
SignedCall(retValue=, isRetValueMock=true, retType=class com.mapbox.navigation.core.lifecycle.MapboxNavigationAppDelegate, self=, method=setup(NavigationOptionsProvider), args=[com.mapbox.navigation.core.lifecycle.MapboxNavigationApp$$Lambda$220/0x0000000800c3d840@5e2c3382], invocationStr=MapboxNavigationAppDelegate(temporary mock).setup(com.mapbox.navigation.core.lifecycle.MapboxNavigationApp$$Lambda$220/0x0000000800c3d840@5e2c3382))
left matchers: [ofType(NavigationOptions)]
...
```
It appears that when we call `mockkObject(MapboxNavigationApp)`, Mockk generates a temporary mock for `MapboxNavigationAppDelegate` to which we don't have access.

Comment (dzinad) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt:148]: You should use `mockkStatic`, not `mockkObject`. Because the method is annotated as `@JvmStatic`.

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/extensions/MapboxExtendableButtonEx.kt [hunk 2]
```diff
@@ -1,28 +0,0 @@
...
-import com.mapbox.navigation.utils.internal.logE
-
-/**
- * Safely change the style of the receiver [MapboxExtendableButton].
- *
- * This method calls [MapboxExtendableButton.updateStyle] and captures [Resources.NotFoundException].
- *
- * @see [MapboxExtendableButton.updateStyle]
- */
-@ExperimentalPreviewMapboxNavigationAPI
```
Comment (dzinad) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/extensions/MapboxExtendableButtonEx.kt:18]: Why did you remove this file?

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/map/scalebar/MapboxMapScalebarParams.kt [hunk 3]
```diff
@@ -1,101 +0,0 @@
...
-import com.mapbox.navigation.base.internal.extensions.inferDeviceLocale
-
-/**
- * Params describing the map scalebar config.
- *
- * @param enabled true if the scalebar should be shown on the map, false otherwise
- * @param isMetricUnits true if the scale bar is using metric system,
- *  false if the scale bar is using imperial units.
- */
-class MapboxMapScalebarParams private constructor(
```
Comment (dzinad) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/map/scalebar/MapboxMapScalebarParams.kt:15]: Why did you remove this file?

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/NavigationViewTest.kt [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+        assertNull(
+            sut.navigationContext.mapViewOwner
+                .getRegisteredObservers()
+                .firstOrNull { it == observer }
+        )
+    }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `setRouteOptionsInterceptor should use RouteOptionsInterceptor to update RouteOptions`() {
```
Comment (dzinad) [inline libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/NavigationViewTest.kt:231]: We can also check setting the null interceptor.

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/extensions/MapboxExtendableButtonEx.kt (hunk 2 above)
Comment (tomaszrybakiewicz) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/extensions/MapboxExtendableButtonEx.kt:18]: Dead code.

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/map/scalebar/MapboxMapScalebarParams.kt (hunk 3 above)
Comment (tomaszrybakiewicz) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/map/scalebar/MapboxMapScalebarParams.kt:15]: We removed it's usage here https://github.com/mapbox/mapbox-navigation-android/pull/6523 and forgot to delete `MapboxMapScalebarParams`.

**Review by dzinad — APPROVED** [review]

**Comment by abhishek1508** [comment]
@tomaszrybakiewicz Heads up that the CI is [failing](https://app.circleci.com/jobs/github/mapbox/mapbox-navigation-android/115771)

**Comment by abhishek1508** [comment]
Also, as per the original ticket we should add tests for `NavigationViewStyles`. I see the task is marked as complete, but I don't see the tests in the pull request. @tomaszrybakiewicz

**Comment by tomaszrybakiewicz** [comment]
@abhishek1508 Done. Fixed static analysis issues and added missing NavigationViewStyles tests.

**Review by abhishek1508 — APPROVED** [review]
Thank you for addressing the feedback!
